freqtrade/worker.py

```diff
@@ -154,6 +155,9 @@ def _process_running(self) -> None:
 
             logger.exception('OperationalException. Stopping trader ...')
             self.freqtrade.state = State.STOPPED
+        except OperationalError as error:
+            logger.warning(f"Warning: {error}, retrying in half a second...")
+            time.sleep(0.5)
```

Comment:
Uh that's a ... dangerous approach, and will clearly fail to do what you expect it to.
There's a reason we only continue in some rare cases (TemporaryError - but not OperationalException or worse).

while it will will stop the exception from crashing the bot - you will end up with open orders (or filled orders?) on the exchange which the bot does not know about (because it failed to insert them into the database) if this exception really happens, and the bot will therefore also not handle these.

You will not see this in dry-runs - as dry-runs ONLY happen within freqtrade, so the bot will never know about the trade in the first place - but in live this WILL fail.

To me, this is a lot worse than failing (because then you know you need to look at what happened).

Reply:
As we discussed on discord: if you think that is true - and I have no reason now to doubt it, then I believe you must mutex the threads of the software because it's obvious it locks at the sqlite library level when a lot of .all()s are happening on a large database.

I will personally use it for the time being because most crashes happen without trading editing.

Reply:
An alternative to restricting the threads from access, would be to wrap the retrying based on the locking exception (which is done by the simple patch originally posted in the PR) in all cases of flushing or other implicit cases of flushing (e.g. those filter().all() situations) so that your requirement for trading safety is satisfied.

I might attempt to do that for my use but for something that ends up upstream you might want something prettier I guess like removing autoflushing and autocommiting and then carefully using commit() which I don't feel confident myself doing correctly but as you had said that might not alter the db locking anyway.